workbench-core/accounts/src/services/projectService.ts

```diff
@@ -56,20 +62,143 @@ export default class ProjectService {
     }
   }
 
+  // TODO--delete after dynamic Authz
+  private _mockGetUserGroups(): string[] {
+    return [];
+  }
+
   /**
    * List projects
    *
+   * @param request - the optional fields to list projects
+   * @param user - authenticated user listing the projects
    * @returns Project entries in DDB
    */
-  public async listProjects(): Promise<{ data: Project[] }> {
-    const queryParams = {
+  // TODO--add filter support
+  public async listProjects(
+    user: AuthenticatedUser,
+    request?: ListProjectsRequest
+  ): Promise<{ data: Project[]; paginationToken: string | undefined }> {
+    const pageSize = request ? request.pageSize : undefined;
+    let paginationToken = request ? request.paginationToken : undefined;
+    // Get user groups--TODO implement after dynamic AuthZ
+    // const userGroupsForCurrentUser: string[] = await this._dynamicAuthorizationService.getUserGroups(user.id);
+    const userGroupsForCurrentUser: string[] = this._mockGetUserGroups(); // mock so the tests work
+
+    // If no group membership, return
+    if (_.isEmpty(userGroupsForCurrentUser)) {
+      return { data: [], paginationToken: undefined };
+    }
+
+    // If IT Admin, return all with pagination
+    if (userGroupsForCurrentUser.length === 1) {
+      if (userGroupsForCurrentUser[0] === 'ITAdmin') {
+        let queryParams: QueryParams = {
+          key: { name: 'resourceType', value: this._resourceType },
+          index: 'getResourceByCreatedAt',
+          limit: pageSize && pageSize >= 0 ? pageSize : DEFAULT_API_PAGE_SIZE
+        };
+
+        queryParams = addPaginationToken(paginationToken, queryParams);
+
+        const projectsResponse = await this._aws.helpers.ddb.query(queryParams).execute();
+
+        paginationToken = getPaginationToken(projectsResponse);
+
+        return Promise.resolve({
+          data: projectsResponse.Items as unknown as Project[],
+          paginationToken: paginationToken
+        });
+      }
+      // If member of 1 group, get project item
+      if (userGroupsForCurrentUser[0] !== 'ITAdmin') {
```

Comment:
This condition is always true and can be removed

Reply:
got it--I changed it to an `else` statement

Reply:
Ha, neither `else` is needed as you have `return`. Your code looks like this:
```
if (a)
  return ;
// else is not needed because if code is here it's already an else close
// some code
```

Reply:
I know it is not needed, but I think it is neater to clearly have different conditionals to enter each section of code. I will get rid of it though since you don't seem to like that.